USWDS-Changelog: Add guidance page changelog data (Phase 4)

amyleadem opened the pull request:
## Summary
This update adds changelog data to guidance page changelogs. The data added covers updates from March 2021 --> present.

:warning: This PR renames the component changelog data files to follow a common naming structure. This will cause some conflicts with https://github.com/uswds/uswds-site/pull/1895. If possible, we should merge https://github.com/uswds/uswds-site/pull/1895 first.

## Related issue
Closes #1892

## Preview links
- [Accessibility](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/accessibility/)
- [Community](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/about/community/)
- [Contribute](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/about/contribute/)
- [Documentation](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/developers/)
- [Getting started for developers - overview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/getting-started-for-developers/)
- [Getting started for developers - phase 1](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/getting-started/developers/phase-one-install/)
- [Getting started for developers - phase 2](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/getting-started/developers/phase-two-compile/)
- [Getting started for developers - phase 3](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/getting-started/developers/phase-three-customize/)
- [Implementations](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/implementations/)
- [Key benefits](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/about/key-benefits/)
- [Maturity model](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/maturity-model/)
- [Migrating to USWDS 3.0](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/migration/)
- [Packages](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/components/packages/)
- [Roadmap](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/about/product-roadmap/)
- [Security](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/about/security/)
- [Settings](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/documentation/settings/)
- [Tokens - Color](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/design-tokens/color/overview/)
- [Tokens - Color - State](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/design-tokens/color/state-tokens/)
- [Tokens - Spacing](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/design-tokens/spacing-units/)
- [Tokens - Shadow](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/design-tokens/shadow)
- [Tokens - Font family](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/design-tokens/typesetting/font-family/)
- [Utilities](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/utilities)
- [Utilities - Display](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/utilities/display/)
- [Utilities - Layout grid](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/utilities/layout-grid/)
- [Utilities - Height and width](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/utilities/height-and-width/)
- [Utilities - Margin](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/utilities/margin-and-padding/)
- [Utilities - Text styles](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/utilities/text-styles/)
- [Website polices and notices](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-changelog-guidance/about/website-policies-notices/)

## Tasks completed
- Added changelog data for guidance pages 
- Renamed changelog .yml files with page type prefix. This makes the files easier to navigate.
- Added changelog keys to the front matter of the related markdown file. This tells the markdown file where to find the corresponding changelog data.
- Updated the `id` for the "Latest updates" section to simply be "changelog". I saw no need to keep the page key as part of the id.

## Review
- Copyeditors: 
  - For the data entries added to the files in `_data/changelogs`:
      - Proofread the copy for error.
      - Confirm that all entries make sense and are "meaningful" to the user. 
- Developers: 
    - Confirm that the updated file naming convention and changelog keys makes sense to you. 
    - Confirm that the updates to `changelog.html` and `changelog-table.html` work as expected.
    
---

Before opening this PR, make sure you’ve done whichever of these applies to you:

- [x] Confirm that this code follows the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `git pull origin [base branch]` to pull in the most recent updates from your base and check for merge conflicts. (Often, the base branch is `develop`).
- [x] Run `npm test` and confirm that all tests pass.
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.

amyleadem commented on _includes/changelog.html:
```diff
@@ -1,10 +1,10 @@
 {% assign changelogItems = changelog.items %}
 
 {%- if changelog.items or changelog.changelogURL -%}
-  <h2 id="{{ componentKey }}-changelog" class="site-component-section-title">Latest updates</h2>
```
Updated the anchor link to just point to `#changelog`. Using the pageKey didn't feel necessary and would require updates if page titles/keys change.

amyleadem commented on _includes/changelog-table.html:
```diff
@@ -1,6 +1,6 @@
-
 {% assign col1Title = 'Date' %}
 {% assign col1Classes = 'tablet:grid-col' %}
+{% assign col1ClassesMinimal = 'tablet:grid-col-3 desktop:grid-col-2' %}
```
Added custom classes for the minimal (two-column) table display found in documentation page types. 

All other changes in this file are formatting fixes.

mejiaj reviewed (commented):
Added some things to consider after an initial code review. If you'd rather keep the changelog table in a single template let me know and I can review/validate guidance changes.

amyleadem commented on _components/footer/footer.md:
```diff
@@ -32,4 +32,7 @@ variants:
+changelog:
+  key: component-footer
 ---
+
```
Thanks, fixed https://github.com/uswds/uswds-site/pull/1919/commits/079654aaeff722ea7ffcbaf805584c88220e4093

amyleadem commented on _includes/changelog-table.html:
```diff
@@ -1,6 +1,6 @@
-
 {% assign col1Title = 'Date' %}
```
Great suggestion. Added `changelog-table-simple`  in [4497072](https://github.com/uswds/uswds-site/pull/1919/commits/4497072f5b6aff1121ad7515e743783491737b77)

amyleadem commented on _data/changelogs/about-community.yml:
```diff
@@ -0,0 +1,30 @@
+title: Community
```
Thanks for bringing this up.
This was originally more useful, but it is now only there for the future collected changelog displays. 
It is safe to change this to optional and when we start building out collected changelogs, we can rely on the page title to be the default. 
For clarity, I added the data key table to the `CHANGELOG-README` and updated `title` to say optional in [ff765a7](https://github.com/uswds/uswds-site/pull/1919/commits/ff765a7be8c67b0bb9607b3f1c3c27d4c79e4ee8)

amyleadem commented on _includes/changelog-table.html:
```diff
@@ -13,90 +13,98 @@
-        <td data-title="{{ col1Title }}" class="{{ col1Classes }}">
-          {% if item.date %}
-            <span class="font-mono-3xs">{{ item.date }}</span>
+    <tr class="{% if item.section %}tablet:border-top-2px tablet:margin-top-2{% endif %}">
```
Removed `item.section`. It must have been an old artifact. Good catch.

amyleadem commented on _includes/changelog-table.html:
```diff
@@ -13,90 +13,98 @@
+      <th scope="col" class="{{ col4Classes }}">{{ col4Title }}</th>
       {% endif %}
-      <th scope="col" class="{{ col5Classes }}">{{ col5Title }}</th>
+      <th scope="col" class="{% unless changelog.type: == 'documentation' %}{{ col5Classes }}{% endunless %}">
```
Simplified table setup by adding `changelog-table-simple`  in [4497072](https://github.com/uswds/uswds-site/pull/1919/commits/4497072f5b6aff1121ad7515e743783491737b77)

amyleadem commented:
Putting this back in draft mode to add more data items.

amyleadem commented:
@mejiaj I have addressed all of your comments. Please let me know if you need anything else.

@bonnieAcameron I discovered more data and have added it to this PR. If you have any questions, please let me know!

bonnieAcameron commented:
Record of offline conversation: I noticed that the guidance change tables have a Date heading in the first column, but they don't have a Description heading in the second column. The component changelogs do have a Description heading. Amy agrees they should have a Description heading and will update.

amyleadem commented:
@bonnieAcameron Great catch. Pushed up a fix in [5f344d4](https://github.com/uswds/uswds-site/pull/1919/commits/5f344d4b9bea58e46a4c3a2794193668a56ab6f8)

bonnieAcameron commented on _data/changelogs/_CHANGELOG-README.md:
```diff
@@ -1,47 +1,45 @@
+
+## Writing a good summary
+  - Changelog summaries should be written in the past tense, use plain language, and begin with a consistent set of verbs whenever possible:
+    - Examples: "Fixed", "Added", "Improved", "Optimized", "Updated", "Introduced", "Styled"
```
```suggestion
 - Examples: "Fixed", "Added", "Improved", "Optimized", "Updated", "Introduced", "Styled"
```
It looks like the indent is hanging a bit too far on this one

bonnieAcameron replied:
nevermindnevermind I see what you did there.

bonnieAcameron commented on _data/changelogs/_CHANGELOG-README.md:
```diff
@@ -1,47 +1,45 @@
+    - Examples: "Fixed", "Added", "Improved", "Optimized", "Updated", "Introduced", "Styled"
+  - Often, a summary can be pulled from the related release notes or pull request. If needed, these summaries should be adjusted to fit the changelog format.
+  - If a single release item accomplished multiple tasks, break the tasks up into separate data items. The goal is to itemize and highlight user benefits and actions required.
+  - If a change item necessitates user action, make sure that the related pull request clearly states a summary of the action required.
```
```suggestion
  - If a change item requires user action, make sure that the related pull request clearly states a summary of the action required.
```
Plain language edit

bonnieAcameron commented on _data/changelogs/about-roadmap.yml:
```diff
@@ -0,0 +1,17 @@
+changelogURL:
+items:
+  - date: 2022-06-16
+    summary: Simplified content and added link to the roadmap on GitHub.
```
```suggestion
    summary: Simplified content and added link to the public roadmap on GitHub.
```

amyleadem replied:
Updated in [6a391cb](https://github.com/uswds/uswds-site/pull/1919/commits/6a391cbe57e7cfdfa53a441c454aad02d0d3758c)

bonnieAcameron commented on _data/changelogs/docs-getting-started-devs-phase-2.yml:
```diff
@@ -0,0 +1,112 @@
+    githubPr: 1473
+    githubRepo: uswds-site
+  - date: 2022-03-10
+    summary: Replaced `uswds-gulp` with `uswds-compile`.
```
I'm wondering if people will understand that this is a replacement to the text on the page rather than the actual file?  See line 18 for an example of how I would expect this to read if it were just about text:

"Updated text to replace 'uswds-gulp' with 'uswds-compile'.

amyleadem replied:
Updated according to the suggestion in https://github.com/uswds/uswds-site/pull/1919#discussion_r1037584948

Updated in: https://github.com/uswds/uswds-site/pull/1919/commits/afe4b145041a1e2f999bf84ce6cdcf5c6c39f208

bonnieAcameron commented on _data/changelogs/docs-getting-started-devs-phase-3.yml:
```diff
@@ -0,0 +1,76 @@
+changelogURL:
+items:
+  - date: 2022-06-17
+    summary: Replaced `_theme-settings.scss` with `_uswds-theme.scss`.
```
I'm wondering if people will understand that this is a replacement to the text on the page rather than the actual file? See line 18 for an example of how I would expect this to read if it were just about text:

"Updated text to replace `_theme-settings.scss` with `_uswds-theme.scss`."

amyleadem replied:
Updated according to the suggestion in https://github.com/uswds/uswds-site/pull/1919#discussion_r1037584948

Updated in: https://github.com/uswds/uswds-site/pull/1919/commits/afe4b145041a1e2f999bf84ce6cdcf5c6c39f208

bonnieAcameron commented on _data/changelogs/docs-getting-started-devs-phase-3.yml:
```diff
@@ -0,0 +1,76 @@
+    githubPr: 1538
+    githubRepo: uswds-site
+  - date: 2022-03-10
+    summary: Replaced `uswds-gulp` with `uswds-compile`.
```
I'm wondering if people will understand that this is a replacement to the text on the page rather than the actual file? See line 18 for an example of how I would expect this to read if it were just about text:

"Updated text to replace 'uswds-gulp' with 'uswds-compile'.

amyleadem replied:
Updated this and similar entries according to the suggestion in https://github.com/uswds/uswds-site/pull/1919#discussion_r1037584948

Updated in: https://github.com/uswds/uswds-site/pull/1919/commits/afe4b145041a1e2f999bf84ce6cdcf5c6c39f208

bonnieAcameron commented on _data/changelogs/docs-implementations.yml:
```diff
@@ -0,0 +1,122 @@
+    githubRepo: uswds-site
+    versionUswds:
+  - date: 2021-05-19
+    summary: Removed "us_web_design_standards gem".
```
```suggestion
    summary: Removed the outdated text: "us_web_design_standards gem".
```

amyleadem replied:
Took a slightly different approach and referred to these item as entries rather than text. 
Update: https://github.com/uswds/uswds-site/pull/1919/commits/d4d2826c529f910d26e2c5a31bbe1b56a522d33a

bonnieAcameron commented on _data/changelogs/docs-implementations.yml:
```diff
@@ -0,0 +1,122 @@
+    githubRepo: uswds-site
+    versionUswds:
+  - date: 2020-09-11
+    summary: Added "uswds-sf-lightning-community".
```
```suggestion
    summary: Added updated text: "uswds-sf-lightning-community".
```

amyleadem replied:
Added clarity: https://github.com/uswds/uswds-site/pull/1919/commits/d4d2826c529f910d26e2c5a31bbe1b56a522d33a

bonnieAcameron commented on _data/changelogs/docs-implementations.yml:
```diff
@@ -0,0 +1,122 @@
+    githubRepo: uswds-site
+    versionUswds:
+  - date: 2020-08-12
+    summary: Added "tailwindcss-uswds".
```
```suggestion
    summary: Added the text: "tailwindcss-uswds".
```

amyleadem replied:
Added clarity: https://github.com/uswds/uswds-site/pull/1919/commits/d4d2826c529f910d26e2c5a31bbe1b56a522d33a

bonnieAcameron commented on _data/changelogs/docs-implementations.yml:
```diff
@@ -0,0 +1,122 @@
+    githubRepo: uswds-site
+    versionUswds:
+  - date: 2020-07-28
+    summary: Added "react-uswds".
```
```suggestion
    summary: Added the text: "react-uswds".
```

amyleadem replied:
Added clarity: https://github.com/uswds/uswds-site/pull/1919/commits/d4d2826c529f910d26e2c5a31bbe1b56a522d33a

bonnieAcameron commented on _data/changelogs/docs-maturity-model.yml:
```diff
@@ -0,0 +1,16 @@
+changelogURL:
+items:
+  - date: 2022-03-10
+    summary: Replaced `uswds-gulp` with `uswds-compile`.
```
```suggestion
    summary: Replaced the outdated text: `uswds-gulp` with updated: `uswds-compile`.
```

amyleadem replied:
Updated in: https://github.com/uswds/uswds-site/commit/afe4b145041a1e2f999bf84ce6cdcf5c6c39f208

bonnieAcameron commented on _data/changelogs/docs-settings.yml:
```diff
@@ -0,0 +1,296 @@
+    versionUswds:
+  - date: 2021-03-17
+    summary: Added `$theme-prefix-separator`.
+    summaryAdditional: This setting allows you to output any character or string for the separator between the prefix and main class.
```
It's not major, but I'm starting to see a little discrepancy between when summaryAdditional is used. In this example, it's detail followed by more detail. In some examples, like "updated page for clarity," there's less detail for less detail--summaryAdditional isn't used at all in those cases. It might be worth updating the readme to address when to include more detail and how.

amyleadem replied:
I see an inconsistency too. Part of me wonders if we need the level of detail we are giving. I might open up an issue so I can step back and evaluate. Are you comfortable moving forward as-is for now?

bonnieAcameron replied:
Yes! Good idea.

bonnieAcameron commented on _data/changelogs/utilities-margin-padding.yml:
```diff
@@ -0,0 +1,18 @@
+items:
+  - date: 2021-06-16
+    summary: Added more negative margin tokens to margin utilities.
+    summaryAdditional: Now you can use utilities like `margin-neg-4` and `margin-neg-5`.
```
1. This is not major, but this is the first time I'm seeing "you" or any more casual-seeming language in these summaries. My suggestion is to keep it as consistent as possible. 
2. The word "now" means little when the "now" it's referencing happened a year and a half ago. 

recommend: update readme to exclude pronouns and conversational time indicators like "now,"

amyleadem replied:
Good point. Added here: https://github.com/uswds/uswds-site/pull/1919/commits/01706bbe08445a449a0fb106e769d50d908a1bde#diff-0bace2ed80976d536baf0a44e4d722dbe3ac44ea0554e3976aecf0f79be4db81R64

bonnieAcameron commented on _data/changelogs/utilities-margin-padding.yml:
```diff
@@ -0,0 +1,18 @@
+items:
+  - date: 2021-06-16
+    summary: Added more negative margin tokens to margin utilities.
+    summaryAdditional: Now you can use utilities like `margin-neg-4` and `margin-neg-5`.
```
```suggestion
    summaryAdditional: Added utilities like `margin-neg-4` and `margin-neg-5`.
```

amyleadem replied:
Simplified copy: [1eff814](https://github.com/uswds/uswds-site/pull/1919/commits/1eff8141578476d0af927dddf7122f0a1a141ed7)

amyleadem commented:
@bonnieAcameron I believe I have addressed all of your comments. If you have any questions, let me know!

bonnieAcameron reviewed (approved):
A thing of beauty is a joy forever.

amyleadem commented:
@mejiaj Can you take another look at the changes made to the table structure? Wanting to make sure I accounted for everything.

mejiaj reviewed (approved):
Everything looks good. I tested:
- [x] Mobile styles
- [x] Desktop styles
- [x] Data in Changelog tables
